### ISSUE TITLE
On "Open this site in assigned container", current container button doesn't always work

### ISSUE THREAD

**Issue by mcomella** [mozilla/multi-account-containers#1331]
- Is "Firefox will: Never remember history" in the Firefox Preferences/Options under "Privacy & Security > History" selected? Yes/No: No
- Are you using Firefox in a Private Window? Yes/No: No
- Can you see a grayed out but ticked Checkbox with the description "Enable Container Tabs" in the Firefox Preferences/Options under "Tabs"? Yes/No: No
- Multi-Account Containers Version: 6.1.0
- Operating System + Version: macOS 10.13.6
- Firefox Version: 66.0a1 2019-01-27
- Other installed Add-ons + Version + Enabled/Disabled-Status: Some add-ons, not sure if relevant

### Actual behavior
Nothing happens.

### Expected behavior
Tab opens in current, i.e. non-default container

### Steps to reproduce
- Select a page to open in a container (my site is mail.google.com and my container is "work")
- Long-press "+" and open new tab in a different container (mine is "google")
- Go to that site (mail.google.com), "Open this site in assigned container" appears
- Select "Open in X container" (the current container: google for me)

### Notes
- This started happening maybe some time late last year or this month (January)
- It only happens sometimes: sometimes it just works
- I noticed if I wait on that page for long enough, sometimes the buttons will start working again.

**Comment by groovecoder** [comment]
@nkkumawat - you are welcome to take this one to submit a PR.

**Comment by bloodmandarin** [comment]
I have a similar issue, but not sure whether it requires a new "issue". I have assigned containers to certain pages, but the corresponding checkmark does not appear afterwards, so I cannot retract that decision anymore. 

In other words, I want to disable the automatic sorting after initially having accepted it. How?

![always open in](https://user-images.githubusercontent.com/9247936/53636306-01e64900-3c20-11e9-9fa0-ed5eed9bdec0.png)

See: https://github.com/Drive4ik/simple-tab-groups/issues/340

**Comment by stoically** [comment]
Only happens for me (on Linux) with Firefox 66.0b11 (64-bit) Developer Edition - not with Firefox 65.0.1 (64-bit), 66.0b11 (64-bit) Beta or 67.0a1 (2019-03-01) (64-bit) Nightly.

@mcomella Since you seem to be on Nightly could you confirm that this doesn't happen anymore for you with the latest 67.0a1?

The underlying problem when it doesn't work is that `explicitOriginalTarget.id` on the confirm page is randomly set or empty. I've tried to reproduce the issue in isolation with a small test-webextension but in this case it doesn't happen. A workaround is available as PR in #1350 if needed.

**Comment by stoically** [comment]
@bloodmandarin This seems to be unrelated. Could you please open a new issue with steps to reproduce?

**Comment by cgatesman** [comment]
FWIW, I see this on macOS Firefox Developer Edition 66.0b11.  It mostly happens when loading an external link (outside of the browser into a new tab (e.g. from Slack)) to a website assigned to a container.  If you keep trying the button, it eventually works.  It's like there is just some kind of massive delay populating the targets to those buttons.

**Comment by arashpayan** [comment]
I just want to chime in and mention that I'm seeing the same problem over the same period as @mcomella described, except that I am on Ubuntu 18.04. More details:
Firefox version: Nightly v67.0a1 (2019-03-17)
OS: Ubuntu 18.04.2 LTS
Multi-Account Containers version: 6.1.0
Other add ons: `Netflix 1080p` and `uBlock Origin`

FWIW, I reset my Firefox profile about a week ago and the problem still remained.

**Comment by rubcap** [comment]
Same issue here, except that I click the blue button on the right to open the tab in the container I choose before. 

_Steps:_
1. Open a "Work" container tab.
2. Go to mail.google.com and select the checkbox (in the add-on pop-up menu) to open it in "Work" by default.
3. Close the container tab.
4. Go to the same site (mail.google.com) in a normal, non-container tab.
5. "Open this site in assigned container?" appears.
6. Check the box "Remember my decision" and press the blue button "Open in Work container".

Firefox version: Release v66.0
OS: Win10
Multi-Account Containers version: 6.1.0
_Expected behavior:_
mail.google.com is immediately opened in a Work container tab.

_Actual behavior:_
Only after a few clicks on the blue button, mail.google.com is opened in a Work container tab.

**Comment by arashpayan** [comment]
I should clarify — I see the problem occurring regardless of which button I click. It takes several attempts for either one.

**Comment by smichel17** [comment]
For me, it seems to be that the area around the text saying the name of the current container is not clickable; the rest of the button is. For reference, that's the red square:

![image](https://user-images.githubusercontent.com/6563664/55288054-776d4280-53a1-11e9-9f7f-19060bebfd8f.png)

keyword: confirmation (adding because I searched for "confirmation page" first and this didn't come up)

**Comment by cgatesman** [comment]
> For me, it seems to be that the area around the text saying the name of the current container is not clickable; the rest of the button is.

This is exactly right!  Here I thought it was some kind of timing issue, but I see now, that it was actually that bolded text that wasn't clickable!  Nice catch!

**Comment by stoically** [comment]
@smichel17 Nice catch. It seems that the behavior of `explicitOriginalTarget` changed and now also takes into account elements nested inside buttons.

**Comment by francisbarton** [comment]
> For me, it seems to be that the area around the text saying the name of the current container is not clickable; the rest of the button is. For reference, that's roughly the red square:
> ![image](https://user-images.githubusercontent.com/6563664/55288054-776d4280-53a1-11e9-9f7f-19060bebfd8f.png)

I came to this bug report to say "me too" but then I found @smichel17 's solution and I wanted to report that it works for me too (Ff dev aurora 67.0b7 / Windows 10) - it seems like there is a fix! Just need to make the clickable area the same size as the whole button.

**Comment by kendallcorner** [comment]
Hi. We're currently triaging the Multi Account Container issues. This appears to have been fixed, so I'm going to close it. If you don't feel the underlying issue has been resolved, feel free to reopen it with updated steps to reproduce.
